pkgs/native_assets_builder/test_data/treeshaking_native_assets/hook/link.dart

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:native_assets_cli/native_assets_cli.dart';
+
+const packageName = 'treeshaking_native_assets';
+
+void main(List<String> arguments) async {
+  await link(arguments, (config, output) async {
+    final usedSymbols =
+        config.resources.map((resource) => resource.metadata.toString());
+    final dynamicLibrary = config.assets.firstWhere(
+        (asset) => asset.id.endsWith('src/${packageName}_bindings.dart'));
+    final staticLibrary =
+        config.assets.firstWhere((asset) => asset.id.endsWith('staticlib'));
```

Comment:
`asset.id.endsWith('staticlib')`?

Wouldn't we expect to be able to loop over all assets given to this link hook? (We might add an assert that it's exactly one.)
In general the build hook sending assets to a link hook should know what the linker can consume, instead of the linker trying to cherry pick.

Is this a leftover from also generating a dynamic library?